app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+import alert from '../lib/alert';
+
+const ID_PREFIX = 'COLLECTION_COLLABORATORS_PAGE_';
+
+const POSSIBLE_ROLES = [
+  'owner',
+  'collaborator',
+  'contributor',
+  'viewer'
+];
+
+const ROLES_INFO = {
+  owner: {
+    label: 'Owner',
+    description: 'The collection creator. There can only be one.'
+  },
+  collaborator: {
+    label: 'Collaborator',
+    description: 'Collaborators have full access to add and remove subjects from the collection.'
+  },
+  contributor: {
+    label: 'Contributor',
+    description: 'Contributors can only add subjects to the collection.'
+  },
+  viewer: {
+    label: 'Viewer',
+    description: 'Viewers can see this collection even if it\'s private.'
+  }
+};
+
+export class RoleCreator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      creating: null,
+      error: null
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const checkboxes = this.node.querySelectorAll('[name="role"]');
+    const users = this.userSearch.value().map(option => parseInt(option.value, 10));
+    const checkedBoxes = Object.keys(checkboxes).filter(key => checkboxes[key].checked);
+    const roles = checkedBoxes.map(value => checkboxes[value].value);
+
+    this.setState({
+      creating: true,
+      error: null
+    });
+
+    const roleSets = users.map(id =>
+      apiClient.type('collection_roles').create({
+        roles,
+        links: {
+          collection: this.props.collection.id,
+          user: id
+        }
+      })
+    );
+
+    Promise.all((roleSets.map((roleSet) => {
+      return roleSet.save().then(role => role);
+    })))
+      .then((args) => {
+        this.userSearch.clear();
+        Object.keys(checkboxes).map((key) => {
+          checkboxes[key].checked = false;
+        });
+        if (this.props.onAdd) {
+          this.props.onAdd.apply(null, args);
+        }
+      })
+      .catch((error) => {
+        this.setState({ error });
+      })
+      .then(() => {
+        this.setState({ creating: false });
+      });
+  }
+
+  render() {
+    let style;
+    let errorMessage;
+
+    if (this.state.creating) {
+      style = {
+        opacity: 0.5,
+        pointerEvents: 'none'
+      };
+    }
+
+    if (this.state.error) {
+      const rawErrorMessage = this.state.error.toString();
+
+      if (rawErrorMessage.indexOf('No User with id')) {
+        errorMessage = 'That user doesn\'t exist!';
+      } else {
+        errorMessage = 'Error adding user.';
+      }
+    }
+
+    return (
+      <div ref={(node) => { this.node = node; }}>
+        {this.state.error && (
+          <p className="form-help error">{errorMessage}</p>
+        )}
+
+        <form style={style}>
+          <div>
+            <UserSearch ref={(component) => { this.userSearch = component; }} />
+          </div>
+
+          <table className="standard-table">
```

Comment:
This is non-blocking since I know this was in the original, but a table being used here is not semantic (it's using the table for formatting rather than representing tabular data). I think a definition list would work better and use CSS for formatting.